exercises/template/ex02_oo_basics/src/main.cpp

```diff
@@ -2,6 +2,29 @@
  * Skeleton main routine
  */
 
-int main(int, char **)
-{
-}
\ No newline at end of file
+#include <iostream>
+#include <memory>
+#include <vector>
+
+#include "shape.hpp"
+
+
+using namespace std;
+
+int main(int, char **) {
+  vector<unique_ptr<Shape>> shapes;
+  shapes.push_back(unique_ptr<Shape>(new Circle(5)));
+  shapes.push_back(unique_ptr<Shape>(new Square(5)));
+  shapes.push_back(unique_ptr<Shape>(new Rectangle(1, 11)));
+  shapes.push_back(unique_ptr<Shape>(new Triangle(5, 6)));
+
+  ShapeSorter sorter(shapes);
```

Comment:
You will need to call `move` here is you use move semantics in the constructor in the example I provided below.